druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -377,9 +381,10 @@ public DruidTable getDruidTable() {
         // A plan where all extra columns are pruned will be preferred.
         .multiplyBy(
             RelMdUtil.linear(querySpec.fieldNames.size(), 2, 100, 1d, 2d))
-        .multiplyBy(getQueryTypeCostMultiplier());
+        .multiplyBy(getQueryTypeCostMultiplier())
+        // a plan with sort pushed to druid is better than doing sort outside of druid
+        .multiplyBy(Util.last(rels) instanceof Bindables.BindableSort ? 0.1 : 0.2);
```

Comment:
We should use the superclass in case other rules do not generate a Sort with bindable convention, i.e., use _instanceof Sort_ instead of _BindableSort_.

Further, if last operator is not a SortLimit, we should just leave cost as it is, i.e., multiply by _1_ instead of _0.2_.